views/public/javascripts/mall-map.js

```diff
@@ -268,54 +260,129 @@ function mallMapJs(){
     // Handle the info panel back button.
     $('a.back-button').click(function (e) {
         e.preventDefault();
-        $('#info-panel').fadeToggle(200, 'linear');
-        $('#toggle-map-button + .back-button').hide();
+        $('#info-panel-container').fadeToggle(200, 'linear');
+        $('#toggle-map-button + .back-button + .next-button').hide();
     });
 
-    /*
-     * Filter markers.
-     *
-     * This must be called on every form change.
-     */
-    function doFilters() {
-        // Prevent concurrent filter requests.
-        if (jqXhr) {
-            jqXhr.abort()
+    function hexToRgb(hex) {
```

Comment:
This is clever, but seems a little overengineered. Do we need to go this far, or could we set opacity separately?

Reply:
I tried but it will change the opacity all the elements instead of just backgrounds. I checkout the original mall map code and they use the rgba for the background color so I stick with this